### ISSUE TITLE
Unsupported sexp: for

### ISSUE THREAD

**Issue by rubys** [opal/opal#2147]
Opal fails to convert the following program:

```
for i in 0...1
  for j in 0...1
    puts i+j
  end
end
```

The issue seems to be the nested `for` loops; having a single `for` loop is not a problem.

Symptoms:

```
        65: from /usr/local/bin/opal:23:in `<main>'
...
        10: from /var/lib/gems/2.7.0/gems/opal-1.0.5/lib/opal/nodes/definitions.rb:60:in `compile_children'
         9: from /var/lib/gems/2.7.0/gems/opal-1.0.5/lib/opal/nodes/definitions.rb:60:in `each'
         8: from /var/lib/gems/2.7.0/gems/opal-1.0.5/lib/opal/nodes/definitions.rb:61:in `block in compile_children'
         7: from /var/lib/gems/2.7.0/gems/opal-1.0.5/lib/opal/nodes/base.rb:105:in `process'
         6: from /var/lib/gems/2.7.0/gems/opal-1.0.5/lib/opal/compiler.rb:358:in `process'
         5: from /var/lib/gems/2.7.0/gems/opal-1.0.5/lib/opal/nodes/base.rb:49:in `compile_to_fragments'
         4: from /var/lib/gems/2.7.0/gems/opal-1.0.5/lib/opal/nodes/logic.rb:236:in `compile'
         3: from /var/lib/gems/2.7.0/gems/opal-1.0.5/lib/opal/nodes/base.rb:109:in `expr'
         2: from /var/lib/gems/2.7.0/gems/opal-1.0.5/lib/opal/compiler.rb:360:in `process'
         1: from /var/lib/gems/2.7.0/gems/opal-1.0.5/lib/opal/compiler.rb:228:in `error'
test.rb:in `end': Unsupported sexp: for (Opal::SyntaxError)
```

**Comment by hmdne** [comment]
It works in master for me (but doesn't work in 1.0.5). I'll try to bisect what fixed it.

**Comment by hmdne** [comment]
Ok so this bug is a duplicate of #2033 and it was fixed by a commit 0f4752f2ccefd126e7a1fe49145cbb7c4d3c95b9

**Comment by rubys** [comment]
@hmdne I'm confused.  If I'm reading correctly, that fix was made in 2019, and included in the v0.0.1-alpha.1 release?

**Comment by hmdne** [comment]
Well, this seems confusing, but those branches are related to @elia trying to do Node.js packaging. We are long without a proper 1.1 release and 1.0 was released somewhere around 2019-05-12.
